File: post_service/nodejs/models/postModel.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

const commentSchema = new Schema({
    comment_id: { type: Schema.Types.ObjectId, required: true },
    user_id: { type: Schema.Types.ObjectId, required: true },
    content: { type: String, required: true },
    created_at: { type: Date, default: Date.now }
});

const postSchema = new Schema({
    post_id: { type: Schema.Types.ObjectId, required: true },
    user_id: { type: Schema.Types.ObjectId, required: true },
    content: { type: String, required: true },
    media: { type: [String] },
    likes: { type: Number, default: 0 },
    comments: [commentSchema],
    created_at: { type: Date, default: Date.now }
});

const Post = mongoose.model('Post', postSchema);

module.exports = Post;

File: post_service/python/models/postmodel.py
from pymongo import MongoClient
from bson.objectid import ObjectId
from datetime import datetime

client = MongoClient('mongodb://localhost:27017/')
db = client['post_database']
posts = db['posts']

def create_post(user_id, content, media=[], likes=0, comments=[]):
    post = {
        "post_id": ObjectId(),
        "user_id": ObjectId(user_id),
        "content": content,
        "media": media,
        "likes": likes,
        "comments": comments,
        "created_at": datetime.utcnow()
    }
    return posts.insert_one(post)

def add_comment(post_id, user_id, content):
    comment = {
        "comment_id": ObjectId(),
        "user_id": ObjectId(user_id),
        "content": content,
        "created_at": datetime.utcnow()
    }
    return posts.update_one({"post_id": ObjectId(post_id)}, {"$push": {"comments": comment}})
